cdap-hbase-compat-0.98/src/main/java/co/cask/cdap/data2/transaction/coprocessor/hbase98/DefaultTransactionProcessor.java

```diff
@@ -103,19 +103,20 @@ public InternalScanner preCompactScannerOpen(ObserverContext<RegionCoprocessorEn
   @Override
   protected void ensureValidTxLifetime(RegionCoprocessorEnvironment env, OperationWithAttributes op,
                                        @Nullable Transaction tx) throws IOException {
-    if (tx == null) {
+    long txId = HBaseTable.getTransactionId(op, tx);
```

Comment:
so is this the write pointer or the tx id?

Reply:
It should be the tx id as write pointer can change during checkpointing. I'll add a comment to that effect.